src/binary/cursor.rs

```diff
@@ -979,6 +970,24 @@ where
         }
     }
 
+    fn read_ivm(&mut self) -> IonResult<StreamItem> {
+        let (major, minor) = self.read_slice(3, |bytes| match *bytes {
+            [major, minor, 0xEA] => Ok((major, minor)),
+            [major, minor, other] => decoding_error(format!(
+                "Illegal IVM: 0xE0 0x{:X} 0x{:X} 0x{:X}",
+                major, minor, other
+            )),
+            _ => unreachable!(), // read_slice will always return the requested number of bytes
```

Comment:
If you put the comment inside the `unreachable!()`, it will be surfaced as part of the ensuing panic if this line of code ever _does_ happen unexpectedly, making it easier to research:
```suggestion
            _ => unreachable!("read_slice did not return the requested number of bytes"),
```